### PR TITLE
Check related topics existence

### PR DESCRIPTION
**Pull request by shifucun**

### CODE REVIEW

File: server/lib/explore/related.py [hunk 1]
```diff
@@ -89,6 +89,24 @@ def compute_related_things(state: ftypes.PopulateState,
       # We found a topic, so break!
       break
 
+  # Check the data existence for related topics
+  all_topics = ([x['dcid'] for x in related_things['parentTopics']] +
```
Comment (pradh) [inline server/lib/explore/related.py:93]: nit: maybe pass it through a set() to dedupe?

File: server/lib/explore/related.py [hunk 2]
```diff
@@ -89,6 +89,24 @@ def compute_related_things(state: ftypes.PopulateState,
       # We found a topic, so break!
       break
 
+  # Check the data existence for related topics
```
Comment (pradh) [inline server/lib/explore/related.py:92]: Maybe move this to a helper function?

File: server/integration_tests/test_data/e2e_edge_cases/emissionsinhouston/chart_config.json [hunk 3]
```diff
@@ -623,57 +623,7 @@
   ],
   "relatedThings": {
     "childPlaces": {},
-    "childTopics": [
```
Comment (pradh) [inline server/integration_tests/test_data/e2e_edge_cases/emissionsinhouston/chart_config.json:626]: Argh, we might need to check fallback place in certain cases.

https://github.com/datacommonsorg/website/blob/a3aaccfc8236dc9eaa82a2e46baaa13b834cbd2a/server/routes/nl/helpers.py#L187-L193

If you want to defer to me that's OK.

File: server/integration_tests/test_data/fulfillment_api_sdg/chart_config.json [hunk 4]
```diff
@@ -1109,57 +1109,7 @@
...
-          "Topic"
-        ]
-      },
-      {
-        "dcid": "dc/topic/sdg_1.b",
-        "name": "Poverty Eradication Policy Frameworks",
-        "types": [
-          "Topic"
-        ]
-      }
```
Comment (pradh) [inline server/integration_tests/test_data/fulfillment_api_sdg/chart_config.json:1161]: Ohh, this is a problem for SDG topics which aren't in the KG!  So we should not do existence check if `dc=sdg`

File: server/lib/explore/related.py [hunk 5]
```diff
@@ -89,6 +89,9 @@ def compute_related_things(state: ftypes.PopulateState,
       # We found a topic, so break!
       break
 
+  if dc != "sdg":
```
Comment (pradh) [inline server/lib/explore/related.py:92]: `if params.is_sdg(state.uttr.insight_ctx):` ?

Because it might be sdg or sdgmini

**Review by pradh — APPROVED** [review]
Thanks Bo!

File: server/lib/explore/related.py (hunk 5 above)
Comment (shifucun) [reply server/lib/explore/related.py:92]: Done

**Comment by shifucun** [comment]
thanks for review!

File: .vscode/launch.json [hunk 6]
```diff
@@ -1,58 +1,53 @@
...
-            ],
-            "cwd": "${workspaceRoot}",
-            "envFile": "${workspaceFolder}/.env",
-            "jinja": true,
-            "console": "internalConsole",
-        }
-    ]
-}
\ No newline at end of file
+  "version": "0.2.0",
```
Comment (pradh) [inline .vscode/launch.json:2]: Do you intend to change this?

File: .vscode/launch.json [hunk 7]
```diff
@@ -1,58 +1,53 @@
...
+      "request": "launch",
+      "stopOnEntry": false,
+      "python": "${workspaceFolder}/.env/bin/python3",
+      "program": "${workspaceRoot}/web_app.py",
+      "env": {
+        "FLASK_APP": "${workspaceRoot}/web_app.py",
+        "FLASK_ENV": "local",
+        "FLASK_DEBUG": "1",
+        "ENABLE_MODEL": "true",
+        "GOOGLE_CLOUD_PROJECT": "datcom-webiste-dev"
```
Comment (pradh) [inline .vscode/launch.json:25]: typo

File: .vscode/launch.json (hunk 6 above)
Comment (shifucun) [reply .vscode/launch.json:2]: Yes, I found my local debug complains about gcloud project, adding this fix the issue.

File: .vscode/launch.json (hunk 7 above)
Comment (shifucun) [reply .vscode/launch.json]: Fixed

**Pull request merged**